tools/llvm-kompile-compute-loc/main.cpp

```diff
@@ -0,0 +1,102 @@
+#include <kllvm/ast/AST.h>
+#include <kllvm/parser/KOREParser.h>
+
+#include <llvm/Support/CommandLine.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace llvm;
+using namespace kllvm;
+
+cl::OptionCategory loc_cat("llvm-kompile-compute-loc options");
+
+cl::opt<std::string> kompiled_dir(
+    cl::Positional, cl::desc("<kompiled-dir>"), cl::Required, cl::cat(loc_cat));
+
+cl::opt<std::string> ordinal(
+    cl::Positional, cl::desc("<ordinal>"), cl::Required, cl::cat(loc_cat));
+
+cl::opt<bool> is_k_line(
+    cl::Positional, cl::desc("[is-k-line]"), cl::init(false), cl::cat(loc_cat));
+
+int64_t get_location(kore_axiom_declaration *axiom) {
+  auto *location_att
+      = axiom->attributes().get(attribute_set::key::Location).get();
+  assert(location_att->get_arguments().size() == 1);
+
+  auto *str_pattern = dynamic_cast<kore_string_pattern *>(
+      location_att->get_arguments()[0].get());
+  std::string location = str_pattern->get_contents();
+
+  size_t l_paren = location.find_first_of('(');
+  size_t first_comma = location.find_first_of(',');
+  size_t length = first_comma - l_paren - 1;
+  return std::stoi(location.substr(l_paren + 1, length));
+}
+
+int64_t get_k_location(std::string &definition) {
+  // Parse the definition.kore to get the AST.
+  kllvm::parser::kore_parser parser(definition);
+  auto kore_ast = parser.definition();
+  kore_ast.get()->preprocess();
+
+  // Iterate through axioms.
+  auto *axiom = kore_ast.get()->get_axiom_by_ordinal(std::stoi(ordinal));
+  if (axiom) {
+    auto start_line = get_location(axiom);
+    return start_line;
+  }
+
+  return -1;
+}
+
+// trim the string from the start
+inline void trim(std::string &s) {
```

Comment:
All right then! .Just because it is a small and self-contained function, that is only a function because it would make the code cleaner. But I can remove the attribute!